jina/jaml/parsers/executor/legacy.py

```diff
@@ -106,8 +106,11 @@ def is_tail_or_head(data: Dict) -> bool:
         :param data: the data for the parser
         :return: True if it is tail/head, False otherwise
         """
-        name = data.get('runtime_args', {}).get('name', '')
-        return 'head' in name or 'tail' in name
+        try:
+            name = data.get('runtime_args', {}).get('name', '')
+            return 'head' in name or 'tail' in name
+        except Exception as _:
```

Comment:
KeyError? Exception seems to broad?

Reply:
It can also be `IterableError`, because they key can be there, but it can be `None`

Reply:
then? KeyError, IterableError ? :P